driver/restdriver.go

```diff
@@ -38,18 +59,261 @@ func (driver *RestDriver) Initialize(logger logger.LoggingClient, asyncValues ch
 }
 
 // HandleReadCommands triggers a protocol Read operation for the specified device.
-func (driver *RestDriver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties, reqs []dsModels.CommandRequest) ([]*dsModels.CommandValue, error) {
-	return nil, fmt.Errorf("RestDriver.HandleReadCommands; read commands not supported")
+func (driver *RestDriver) HandleReadCommands(deviceName string, protocols map[string]models.ProtocolProperties, reqs []dsModels.CommandRequest) (responses []*dsModels.CommandValue, err error) {
+
+	var reading interface{}
+	var val interface{}
+	var result = &dsModels.CommandValue{}
+	var uri string
+	var protocolParams RestProtocolParams
+	responses = make([]*dsModels.CommandValue, len(reqs))
+
+	// To send request to any end device, first we need to know end device details.
+	// Such as end device IP address, port number on which REST server is running etc.
+	// First get all these details from the device file
+	err = getDeviceParameters(&protocolParams, protocols)
```

Comment:
function should return `RestProtocolParams`
```suggestion
	protocolParams , err = getDeviceParameters(protocols)
```